### PR TITLE
Add Morph OCaml framework

### PR DESCRIPTION
**Pull request by ulrikstrid**
Adding the Morph framework which is based on Httpaf instead of Cohttp and is therefore hopefully slightly faster than webmachine.

I couldn't run the tests locally because of #6065

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -65,6 +65,8 @@ env:
     - "TESTLANG=Nim"
     - 'TESTDIR="Nim/httpbeast Nim/jester Nim/basolato"'
     - "TESTLANG=OCaml"
+    - "TESTDIR=OCaml/morph"
```
Comment (mudrz) [inline .travis.yml:68]: hey @ulrikstrid explicitly listing the testdir does not seem to be necessary - for example https://github.com/TechEmpower/FrameworkBenchmarks/blob/30bb2a4047fd24475392f746faef03133e0c5c2e/.travis.yml#L15-L16

(my PR for opium was also merged yesterday, so the list is not full)

File: frameworks/OCaml/morph/morph.dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+FROM reasonnative/web:4.10.0 as builder
+
+RUN mkdir /app
+WORKDIR /app
+
+COPY src/esy.json src/morph-tfb.opam /app/
+
+RUN esy install
+RUN esy build --release
```
Comment (mudrz) [inline frameworks/OCaml/morph/morph.dockerfile:12]: is this esy build intended?

File: frameworks/OCaml/morph/morph.dockerfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+COPY ./src/bin /app/src/bin
+
+RUN esy install
```
Comment (mudrz) [inline frameworks/OCaml/morph/morph.dockerfile:16]: is the second esy install intended?

File: frameworks/OCaml/morph/morph.dockerfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+RUN esy mv "#{self.target_dir / 'default' / 'bin' / 'tfb.exe'}" main.exe
+
+RUN strip main.exe
+
+FROM scratch as runtime
+
+ENV OPENSSL_STATIC=1
+ENV SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
+ENV SSL_CERT_DIR=/etc/ssl/certs
+COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
```
Comment (mudrz) [inline frameworks/OCaml/morph/morph.dockerfile:29]: are these certs automatically used by morph?

**Comment by ulrikstrid** [comment]
Thanks for the review @mudrz I have made a lot of changes, now and I think it's better.
I didn't get static linking working with postgres which I hoped when I started, but this is probably a good start at least.

**Comment by rbjorklin** [comment]
@ulrikstrid looks like the CI might have had a temporary network problem. Try pushing a dummy commit to re-trigger the build?

```
Step 5/9 : RUN apt-get install -yqq siege

 ---> Running in 12799899e565

E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/s/siege/siege_4.0.4-1build1_amd64.deb  Could not connect to archive.ubuntu.com:80 (91.189.88.142), connection timed out
```

**Comment by ulrikstrid** [comment]
Thus should be good to go now

**Comment by rbjorklin** [comment]
@ulrikstrid It looks like Morph failed to start. See the [detailed view](https://tfb-status.techempower.com/results/223b29a9-851d-4f0d-9b3c-bb5b89d65d97) for the first test run that included Morph.
* [morph.log](https://tfb-status.techempower.com/unzip/results.2020-10-28-23-03-48-389.zip/results/20201024111848/morph/build/morph.log)
* [morph-static.log](https://tfb-status.techempower.com/unzip/results.2020-10-28-23-03-48-389.zip/results/20201024111848/morph-static/build/morph-static.log)

**Comment by ulrikstrid** [comment]
Thank you, I didn't know where to find the logs. I should be able to fix this my running the nightly of esy.

**Comment by ulrikstrid** [comment]
Hopefully fixed in #6110

**Comment by mudrz** [comment]
@ulrikstrid  it has worked https://www.techempower.com/benchmarks/#section=test&runid=10218082-6e18-4513-9be4-e721bb8a1596&hw=ph&test=json&l=zik0zj-v&a=2

**Pull request merged**